### ISSUE TITLE
/GET rest api can not be performed when /config api is called

### ISSUE THREAD

**Issue by bjzhangl** [zalando/patroni#2030]
I am using patroni 2.1.0 and want to change max_locks_per_transaction parameter. 
First I call /config api to change dcs and reload parameter for postgresql. And then call /GET api to check if pending_restart is set. If it is set, /restart api will be called. 
This works well for patroni 1.6.5. But failed in 2.1.0.
This is the log after /config api returns.
```
172.31.0.1 - - [11/Aug/2021:09:53:27] "PUT /api/v1/dbserver/configuration HTTP/1.1" 200 47 "" "python-requests/2.22.0"
2021-08-11 09:53:27,617 patroni INFO: no action. I am (35dc5e4f-fd6e-4148-839e-5d614d31b5da_mMVgu3D8) the leader with the lock
2021-08-11 09:53:27,833 db-agent DEBUG system_utils.py:410 API Actions POST received arguments: start, body: {'operation': 'dynamic_restart_primary'}.
2021-08-11 09:53:27,833 db-agent DEBUG service.py:106 Waiting for pending restart to be set.
2021-08-11 09:53:27,835 db-agent ERROR patroni_api.py:63 Get node status from patroni failed: HTTPSConnectionPool(host='127.0.0.1', port=8008): Max retries exceeded with url: /patroni (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x3ff892c63a0>: Failed to establish a new connection: [Errno 104] Connection reset by peer'))
2021-08-11 09:53:27,843 patroni INFO: Changed max_locks_per_transaction from 500 to 550 (restart might be required)
2021-08-11 09:53:27,843 patroni INFO: Reloading PostgreSQL configuration.
server signaled
2021-08-11 09:53:27,865 db-agent ERROR system_utils.py:419 Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 159, in _new_conn
    conn = connection.create_connection(
  File "/usr/lib/python3/dist-packages/urllib3/util/connection.py", line 84, in create_connection
    raise err
  File "/usr/lib/python3/dist-packages/urllib3/util/connection.py", line 74, in create_connection
    sock.connect(sa)
ConnectionResetError: [Errno 104] Connection reset by peer

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 665, in urlopen
    httplib_response = self._make_request(
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 376, in _make_request
    self._validate_conn(conn)
  File "/usr/lib/python3/dist-packages/urllib3/connectionpool.py", line 996, in _validate_conn
    conn.connect()
  File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 314, in connect
    conn = self._new_conn()
  File "/usr/lib/python3/dist-packages/urllib3/connection.py", line 171, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.VerifiedHTTPSConnection object at 0x3ff892c63a0>: Failed to establish a new connection: [Errno 104] Connection reset by peer

```
This is the setting  of patroni.yml.
```
bootstrap:
  dcs:
    loop_wait: 10
    maximum_lag_on_failover: 1048576
    postgresql:
      parameters:
        max_connections: 115
        max_locks_per_transaction: 550
      use_pg_rewind: true
    retry_timeout: 30
    ttl: 70
  initdb:
  - encoding: UTF8
  - data-checksums
  - auth-host: md5
  - auth-local: trust
etcd:
  cacert: /etc/ssl/etcd/ca.crt
  cert: /etc/ssl/etcd/client.crt
  hosts:
  - 172.16.68.2:2379
  - 172.16.132.2:2379
  - 172.16.68.3:2379
  - 172.16.132.3:2379
  - 172.16.132.4:2379
  key: /etc/ssl/etcd/client.key
  protocol: https
name: 35dc5e4f-fd6e-4148-839e-5d614d31b5da_mMVgu3D8
namespace: /psql-cluster/
postgresql:
  authentication:
    replication:
      password: 0hSW1fhVu9CzlgB5w8OXFt8xVSnQWcrnIGZ5
      username: replicator
    superuser:
      password: TB5iqHSANHpaSUHbX4VSZv0aNbdtVsq7SEWo
      username: postgres
  bin_dir: /usr/lib/postgresql/13/bin/
  config_dir: /data/share/conf
  connect_address: 172.16.68.3:20179
  data_dir: /data/db
  listen: 0.0.0.0:5432
  pgpass: /data/share/conf/.pgpass
  stop_timeout: 60
restapi:
  certfile: /etc/ssl/private/dbaas.crt
  keyfile: /etc/ssl/private/dbaas.key
  listen: 0.0.0.0:8008
  verify_client: none
scope: 35dc5e4f-fd6e-4148-839e-5d614d31b5da
tags:
  clonefrom: false
  nofailover: false
  noloadbalance: false
  nosync: false
watchdog:
  mode: false
```
I do not know what led connection can not be set up. My setting is not enough?
I use this code to call rest api:
```
resp = requests.get(self.get_url, verify = False)
```

**Comment by CyberDem0n** [comment]
Sorry, but I only see some errors from your scripts, not from Patroni.
Besides that, what you claim that doesn't work is covered with behave tests and executed for every commit: https://github.com/zalando/patroni/blob/master/features/patroni_api.feature#L37-L48
Aslo, there is the `restapi.connect_address` in your config?

**Comment by bjzhangl** [comment]
@CyberDem0n there is no restapi.connect_address. My confusing is that /config can be called correctly. But when calling /get right after /config, connection is reset by peer. Is there any setting i need to change from 1.6.5 to 2.1.0?

**Comment by CyberDem0n** [comment]
You probably facing the same issue as here https://github.com/zalando/patroni/issues/2003 (already fixed in the master branch).

But, independent from it, the `restapi` section **MUST** have the meaningful value of the `connect_address`. If the `connect_address` isn't defined it will use the value from `listen`. The `connect_address` **MUST** be accessible from other nodes in the Patroni cluster. Please read the documentation: https://patroni.readthedocs.io/en/latest/SETTINGS.html#rest-api. With 0.0.0.0:8008 the automatic failover is crippled.

**Comment by bjzhangl** [comment]
@CyberDem0n I look through #2003. The error reported is not the same with mine. Mine is error 104. But still I will apply the patch and check again.
I make an experiment, call /get 5s after /config but not immediately. The error disappears. Everything works well.
```
2021-08-12 05:52:39,712 patroni INFO: Changed max_locks_per_transaction from 550 to 560 (restart might be required)
2021-08-12 05:52:39,713 patroni INFO: Reloading PostgreSQL configuration.
server signaled
2021-08-12 05:52:44,538 db-agent DEBUG service.py:102 Get node status from patroni: {'state': 'running', 'postmaster_start_time': '2021-08-12 05:40:09.948067+00:00', 'role': 'master', 'server_version': 130003, 'cluster_unlocked': False, 'xlog': {'location': 68407424}, 'timeline': 5, 'replication': [{'usename': 'replicator', 'application_name': '35dc5e4f-fd6e-4148-839e-5d614d31b5da_EMtno9OB', 'client_addr': '172.16.132.4', 'state': 'streaming', 'sync_state': 'async', 'sync_priority': 0}, {'usename': 'replicator', 'application_name': '35dc5e4f-fd6e-4148-839e-5d614d31b5da_ftR9EaB7', 'client_addr': '172.16.132.3', 'state': 'streaming', 'sync_state': 'async', 'sync_priority': 0}], 'database_system_identifier': '6995088096876790006', 'pending_restart': True, 'patroni': {'version': '2.1.0', 'scope': '35dc5e4f-fd6e-4148-839e-5d614d31b5da'}}
```

**Comment by bjzhangl** [comment]
@CyberDem0n I applied fix in #2003, it can fix my problem. Thanks!
